### PR TITLE
Rover: implement external control

### PR DESCRIPTION
**Pull request by srmainwaring**
Implement external control for Rover.

### Testing

Tested in Gazebo with WildThumper robot, cartographer and navigation 2.

rover-ext-ctrl

### CODE REVIEW

File: Rover/AP_ExternalControl_Rover.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      Velocity is in earth frame, NED [m/s].
+      Yaw is in earth frame, NED [rad/s].
+     */
+    bool set_linear_velocity_and_yaw_rate(const Vector3f &linear_velocity, float yaw_rate_rads) override;
+private:
+    /*
+      Return true if Copter is ready to handle external control data.
+      Currently checks mode and arm states.
+    */
+    bool ready_for_external_control();
```
Comment (Ryanf55) [inline Rover/AP_ExternalControl_Rover.h:23]: Could add `WARN_IF_UNUSED` for these two functions

Comment (srmainwaring) [reply Rover/AP_ExternalControl_Rover.h:23]: Could do. They're not marked as such in the Copter implementation of external control. What's the recommended house style for adding these attributes?

Comment (Ryanf55) [reply Rover/AP_ExternalControl_Rover.h:23]: I'd add them to Copter, perhaps in a separate PR. It's just additional compile-time-safety, not critical or a blocker.

File: Rover/AP_ExternalControl_Rover.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+public:
+    /*
+      Set linear velocity and yaw rate. Pass NaN for yaw_rate_rads to not control yaw.
+      Velocity is in earth frame, NED [m/s].
+      Yaw is in earth frame, NED [rad/s].
+     */
+    bool set_linear_velocity_and_yaw_rate(const Vector3f &linear_velocity, float yaw_rate_rads) override;
+private:
+    /*
+      Return true if Copter is ready to handle external control data.
```
Comment (nexton-winjeel) [inline Rover/AP_ExternalControl_Rover.h:20]:
```suggestion
      Return true if Rover is ready to handle external control data.
```

Comment (srmainwaring) [reply Rover/AP_ExternalControl_Rover.h]: Thanks for the catch @nexton-winjeel

File: Rover/AP_ExternalControl_Rover.h (hunk 1 above)
Comment (srmainwaring) [reply Rover/AP_ExternalControl_Rover.h:23]: Ok - maybe we could do as a follow up in both cases to keep things consistent.

**Review by Ryanf55 — APPROVED** [review]

**Review by IamPete1 — APPROVED** [review]
LGTM, looks very like copter implementation.

**Pull request merged**